src/vs/platform/windows/electron-main/windowsMainService.ts

```diff
@@ -847,7 +855,8 @@ export class WindowsMainService extends Disposable implements IWindowsMainServic
 				pathsToOpen.push(path);
 			}
 		}
-		return pathsToOpen;
+		const profileName = cli['settings-profile'];
```

Comment:
Hm, what if I pass CLI option but not workspaces to open?

Reply:
It is ignored. This option has to be used always with a workspace as they are workspace associated.